Fix dman not delivering tool (but picking them up still)

xavierh opened the pull request:
In testing/1.11 the dman get the tools from the warehouse but does not deliver them
Review please

xavierh committed:
Fix dman not delivering tool (but picking them up still)

MinecoloniesSonar commented:
SonarQube analysis reported 29 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 26 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L32](https://github.com/Minecolonies/minecolonies/blob/2b582ccdf1e3eb95bfece5135b7836b4d30715dd/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L32): Reduce the number of "static" imports in this class from 5 to the maximum allowed 4. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3030)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L503](https://github.com/Minecolonies/minecolonies/blob/2b582ccdf1e3eb95bfece5135b7836b4d30715dd/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L503): Reduce the number of returns of this method 7, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L503](https://github.com/Minecolonies/minecolonies/blob/2b582ccdf1e3eb95bfece5135b7836b4d30715dd/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L503): Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/2b582ccdf1e3eb95bfece5135b7836b4d30715dd/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/2b582ccdf1e3eb95bfece5135b7836b4d30715dd/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/2b582ccdf1e3eb95bfece5135b7836b4d30715dd/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/2b582ccdf1e3eb95bfece5135b7836b4d30715dd/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/2b582ccdf1e3eb95bfece5135b7836b4d30715dd/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/2b582ccdf1e3eb95bfece5135b7836b4d30715dd/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/2b582ccdf1e3eb95bfece5135b7836b4d30715dd/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

xavierh committed:
rename isToolInHut -> retrieveToolInHut
rename isToolInTileEntity -> retrieveToolInTileEntity
quit early is the tool Type is NONE

MinecoloniesSonar commented:
SonarQube analysis reported 37 issues
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 30 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L40](https://github.com/Minecolonies/minecolonies/blob/02fdaad1e696f9f7cc8b05fafe45fcbe29e87e2b/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L40): class "AbstractEntityAIBasic" has 55 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L205](https://github.com/Minecolonies/minecolonies/blob/02fdaad1e696f9f7cc8b05fafe45fcbe29e87e2b/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L205): Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L533](https://github.com/Minecolonies/minecolonies/blob/02fdaad1e696f9f7cc8b05fafe45fcbe29e87e2b/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L533): Remove this unused private "requestWithoutSpam" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L987](https://github.com/Minecolonies/minecolonies/blob/02fdaad1e696f9f7cc8b05fafe45fcbe29e87e2b/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L987): Provide multiple methods instead of using "useItemDamage" to determine which action to take. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L1036](https://github.com/Minecolonies/minecolonies/blob/02fdaad1e696f9f7cc8b05fafe45fcbe29e87e2b/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L1036): Provide multiple methods instead of using "useItemDamage" to determine which action to take. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L32](https://github.com/Minecolonies/minecolonies/blob/02fdaad1e696f9f7cc8b05fafe45fcbe29e87e2b/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L32): Reduce the number of "static" imports in this class from 5 to the maximum allowed 4. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3030)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L503](https://github.com/Minecolonies/minecolonies/blob/02fdaad1e696f9f7cc8b05fafe45fcbe29e87e2b/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L503): Reduce the number of returns of this method 7, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/02fdaad1e696f9f7cc8b05fafe45fcbe29e87e2b/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/02fdaad1e696f9f7cc8b05fafe45fcbe29e87e2b/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/02fdaad1e696f9f7cc8b05fafe45fcbe29e87e2b/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

Raycoms commented:
:shipit: as well

[![Approved with PullApprove](https://img.shields.io/badge/maintainers-approved-brightgreen.svg)](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1214/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)
